portfolio/src/main/java/com/google/sps/servlets/NewTaskServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Servlet responsible for creating new tasks. */
+@WebServlet("/new-task")
+public class NewTaskServlet extends HttpServlet {
```

Comment:
Instead of creating separate endpoints `/new-task`, `/delete-task`, and `list-task`, you can actually just put it all under one endpoint and just use different HTTP methods on that endpoint.

This means that you really only need one servlet for this which manages everything related to the Task resource.

For example, you can create a servlet called `TaskServlet` at the endpoint `/task` and in this servlet, you can have the methods `doGet` which returns all tasks, `doPost` which creates a new task, and `doDelete` which deletes a task.

Then, in your frontend code, all you'd need to do is specify GET, POST, or DELETE requests for each of these scenarios.

Reply:
Good to know. Three classes for Post, Get Delete must be redundant. Made TaskServlet have doPost, doGet, and doDelete.